models/contact.js

```diff
@@ -1,3 +1,66 @@
+/* eslint-disable no-underscore-dangle */
+const Sequelize = require('sequelize');
+
+const { Op } = Sequelize;
+
+const getPreviousAndNewCases = async (contactInstance, transaction) => {
+  const { Case } = contactInstance.sequelize.models;
+  const previousCaseId = contactInstance._previousDataValues.caseId;
```

Comment:
I didn't know about `previous`. I'll investigate to see if it handles everything we need.

Reply:
so, `previous(key)` will return the previous value for that `key`, and `previous()` without argument will return the previous values that have changed. There's no direct substitution for `_previousDataValues`, but we can easily achieve that with `{...contactInstance.dataValues, ...contactInstance.previous()}`. I'll make the change, just in case they endup making `_previousDataValues` unaccessible in the future.

Reply:
That sounds good to me, but one thought:  could doing `{...contactInstance.dataValues, ...contactInstance.previous()}` mean that we end up including a value that didn't previously exist?  Like, let's say we added `"foo": "bar"` and it's now in the `dataValues`.  Would `previous()` return `"foo": undefined` and then it would be omitted?

Reply:
I actually don't know how `previous()` would behave in this situation, and it may lead to this issue of including a value that shouldn't exist. In practice, this is currently unlikely to happen because `Case` will have all properties set in its creation. But now I feel our code could be very vulnerable in the future because of this. 

Let me try to test this scenario and see what happens in this situation. I'll get back with an answer.

Reply:
I've investigated it a little bit more, and it behave correclty: `previous()` brought `newProp: null`. I believe it's safe to go on with this approach.